src/citrine/informatics/predictors/expression_predictor.py

```diff
@@ -0,0 +1,194 @@
+from typing import Optional, Mapping
+from warnings import warn
+
+from citrine._serialization import properties as _properties
+from citrine._serialization.serializable import Serializable
+from citrine._session import Session
+from citrine.informatics.descriptors import Descriptor, RealDescriptor
+from citrine.informatics.reports import Report
+from citrine.informatics.predictors import Predictor
+
+__all__ = ['ExpressionPredictor', 'DeprecatedExpressionPredictor']
+
+
+class ExpressionPredictor(Serializable['ExpressionPredictor'], Predictor):
+    """A predictor that computes an output from an expression and set of bounded inputs.
+
+    .. seealso::
+       If you are using the deprecated predictor please see
+       :class:`~citrine.informatics.predictors.DeprecatedExpressionPredictor` for an example that
+       shows how to migrate to the new format.
+
+    Parameters
+    ----------
+    name: str
+        name of the configuration
+    description: str
+        the description of the predictor
+    expression: str
+        expression that computes an output from aliased inputs
+    output: RealDescriptor
+        descriptor that represents the output relation
+    aliases: Mapping[str, RealDescriptor]
+        a mapping from each unknown argument to its descriptor.
+        All unknown arguments must have an associated descriptor.
+
+    """
+
+    expression = _properties.String('config.expression')
+    output = _properties.Object(RealDescriptor, 'config.output')
+    aliases = _properties.Mapping(_properties.String, _properties.Object(RealDescriptor),
+                                  'config.aliases')
+    typ = _properties.String('config.type', default='AnalyticExpression', deserializable=False)
+
+    # NOTE: These could go here or in _post_dump - it's unclear which is better right now
+    module_type = _properties.String('module_type', default='PREDICTOR')
+
+    def __init__(self,
+                 name: str,
+                 description: str,
+                 expression: str,
+                 output: RealDescriptor,
+                 aliases: Mapping[str, RealDescriptor],
+                 session: Optional[Session] = None,
+                 report: Optional[Report] = None,
+                 archived: bool = False):
+        self.name: str = name
+        self.description: str = description
+        self.expression: str = expression
+        self.output: Descriptor = output
+        self.aliases: Mapping[str, RealDescriptor] = aliases
+        self.session: Optional[Session] = session
+        self.report: Optional[Report] = report
+        self.archived: bool = archived
+
+    def _post_dump(self, data: dict) -> dict:
+        data['display_name'] = data['config']['name']
+        return data
+
+    def __str__(self):
+        return '<ExpressionPredictor {!r}>'.format(self.name)
+
+
+class DeprecatedExpressionPredictor(Serializable['DeprecatedExpressionPredictor'], Predictor):
```

Comment:
@latture how do we decide when we can remove this?

Reply:
We should follow up with product/DS to see if it's still being used. My guess is that it's probably safe, but we should check and potentially make an announce that it's being removed in the next version first.